src/jit/morph.cpp

```diff
@@ -6953,6 +6958,10 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
         {
             szFailReason = "Localloc used";
         }
+        else if ((call->gtFlags & GTF_CALL_NO_RETURN) != 0)
+        {
+            szFailReason = "No return";
+        }
```

Comment:
This is exactly the kind of thing that I'd like to see in a comment. A pointer to the test case that failed w/o this exclusion would be useful here too.